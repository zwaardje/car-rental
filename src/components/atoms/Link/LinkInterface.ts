import { ReactNode, HTMLAttributes } from 'react';

export interface Link extends HTMLAttributes<any>{
    url: string;
    children: ReactNode | string;
}
