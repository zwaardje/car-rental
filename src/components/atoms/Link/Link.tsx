import * as React from 'react';

import { Link as LinkInterface } from './LinkInterface';

export interface Props extends LinkInterface {}
export interface State {}

export default class Link extends React.Component<Props, State> {
    static displayName = 'Link';

    render() {
        const { children, url, className } = this.props;

        return (
            <a className={`link ${className ? ` ${className}` : ''}`} href={url}>
                {children}
            </a>
        );
    }
}
