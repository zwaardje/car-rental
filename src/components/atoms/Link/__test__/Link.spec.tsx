import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Link from '../Link';

import { defaultProps } from '../__data__/Link.mockdata';

describe('Link', () => {

    it('renders primary link correctly', () => {
        const props = { ... defaultProps };
        const tree = renderer.create(<Link {... props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
