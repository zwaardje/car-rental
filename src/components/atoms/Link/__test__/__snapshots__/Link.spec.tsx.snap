// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders primary link correctly 1`] = `
<a
  className="link link__primary"
  href="#"
/>
`;

exports[`Link renders secondary link correctly 1`] = `
<a
  className="link link__secondary"
  href="#"
/>
`;

exports[`Link renders underline link correctly 1`] = `
<a
  className="link link__underline"
  href="#"
/>
`;
